File: infrastructure/src/main/java/com/study/platform/domain/chat/infrastructure/ChatMessageRepositoryAdapter.java
package com.study.platform.domain.chat.infrastructure;

import com.study.platform.domain.chat.model.ChatMessage;
import com.study.platform.domain.chat.model.ChatMessageRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class ChatMessageRepositoryAdapter implements ChatMessageRepository {

    private final ChatMessageJpaRepository chatMessageJpaRepository;

    @Override
    public ChatMessage save(ChatMessage message) {
        return chatMessageJpaRepository.save(message);
    }
}

File: infrastructure/src/main/kotlin/com/study/platform/domain/chat/infrastructure/ChatMessageRepositoryAdapter.kt
package com.study.platform.domain.chat.infrastructure

import com.study.platform.domain.chat.model.ChatMessage
import com.study.platform.domain.chat.model.ChatMessageRepository
import org.springframework.stereotype.Repository

@Repository
class ChatMessageRepositoryAdapter(
    private val chatMessageJpaRepository: ChatMessageJpaRepository
) : ChatMessageRepository {

    override fun save(message: ChatMessage): ChatMessage =
        chatMessageJpaRepository.save(message)
}
